File: problems/1436-destination-city/solution.cpp
class Solution {
public:
    string destCity(vector<vector<string>>& paths) {
        unordered_set<string> starts;
        for (const auto& path : paths) {
            starts.insert(path[0]);
        }

        for (const auto& path : paths) {
            if (starts.find(path[1]) == starts.end()) {
                return path[1];
            }
        }

        return "";
    }
};

File: problems/1436-destination-city/solution.py
class Solution:
    def destCity(self, paths: List[List[str]]) -> str:
        start_cities = set()
        end_cities = set()
        for path in paths:
            start_cities.add(path[0])
            end_cities.add(path[1])
        for city in end_cities:
            if city not in start_cities:
                return city
        return ""